src/mediarithmics/plugins/common/BasePlugin.ts

```diff
@@ -246,15 +248,30 @@ export abstract class BasePlugin {
     );
   }
 
-  fetchDataFile(uri: string): Promise<Buffer> {
-    return this.requestGatewayHelper(
-      "GET",
-      `${this.outboundPlatformUrl}/v1/data_file/data`,
-      undefined,
-      {uri: uri},
-      false,
-      true
-    );
+  async fetchDataFile(uri: string): Promise<Buffer> {
+
+    // If dataFile is already cached, fetch file only if modified.
+    const headers = this.dataFileCachingDate ? { 'If-Modified-Since': this.dataFileCachingDate } : undefined;
+
+
+    try {
+      const response = await this.requestGatewayHelper(
+        "GET",
+        `${this.outboundPlatformUrl}/v1/data_file/data`,
+        undefined,
+        {uri: uri},
+        false,
+        true,
+        headers,
+      );
+      // Cache timestamp of request and file
+      this.dataFileCachingDate = new Date().toUTCString();
+      this.dataFileCache = response;
+    } catch(e) {
```

Comment:
Could you write a more specific condition to specifically handle the `statusCode=301` condition?

We could have other kind of errors (404 / 500 / etc.) that shouldn't be mixed up with the caching case.